### PR TITLE
Packaging version 2

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: manifest.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+[resources]
+    [resources.system_user]
+
+    [resources.install_dir]
+
+    [resources.data_dir]
+
+    [resources.permissions]
+    main.label = "api"
+    main.url = "re:$domain\/.well-known\/.*"
```
Comment (alexAubin) [inline manifest.toml:63]:
```suggestion
    main.url = "re:$domain\/.well-known\/.*"
```

Not sure why, but TOML is not happy about this :grimacing:  : `Reserved escape sequence used`

File: manifest.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    main.allowed = ["visitors, all_users"]
```
Comment (alexAubin) [inline manifest.toml:64]:
```suggestion
```

This is not relevant considering that there's an `init_main_permission` which is meant to initialize this

File: conf/config_install.json [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-    "datadirectory": "__DATADIR__",
-    "trusted_domains": [
-      "localhost",
-      "__DOMAIN__"
-    ]
-  }
-}
+{
+  "system": {
+    "data_directory": "__DATA_DIR__",
```
Comment (alexAubin) [inline conf/config_install.json:3]:
```suggestion
    "datadirectory": "__DATA_DIR__",
```

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: manifest.toml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    [resources.system_user]
+
+    [resources.install_dir]
+
+    [resources.data_dir]
+
+    [resources.permissions]
+    main.url = "/"
+    
+    api.url = "re:$domain\\/.well-known\\/.*"
```
Comment (alexAubin) [inline manifest.toml:64]:
```suggestion
    api.url = "re:__DOMAIN__\\/.well-known\\/.*"
```

i'm pushing a change in the core to support this syntax

(Also i'm not sure if we really need a regex here or if `__DOMAIN__/.well-known/` could be enough

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!bookwormtestme

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
ready to be tested!

**Pull request closed without merging**